pkg/codegen/testing/test/dotnet.go

```diff
@@ -121,8 +122,13 @@ func checkDotnet(t *testing.T, path string, dependencies codegen.StringSet, pulu
 	if err = os.Remove(programFile); !os.IsNotExist(err) {
 		require.NoError(t, err)
 	}
+	dotnetVersion := "8"
+	if version := os.Getenv("DOTNET_VERSION"); version != "" {
+		dotnetVersion = version
+	}
+	dotnetVersion = fmt.Sprintf("net%s.0", dotnetVersion)
 	err = integration.RunCommand(t, "create dotnet project",
-		[]string{ex, "new", "console", "-f", "net8.0"}, dir, &integration.ProgramTestOptions{})
+		[]string{ex, "new", "console", "-f", dotnetVersion}, dir, &integration.ProgramTestOptions{})
```

Comment:
I think this is an actual improvement compared to before.  Before we would always just use dotnet 8 here, regardless of what we would set in CI.